src/Widgets/ColumnContainerWidget/ColumnContainerWidgetComponent.tsx

```diff
@@ -1,33 +1,76 @@
-import { provideComponent, ContentTag } from 'scrivito'
+import { provideComponent, ContentTag, connect } from 'scrivito'
 import { ColumnContainerWidget } from './ColumnContainerWidgetClass'
+import { ColumnWidgetInstance } from '../ColumnWidget/ColumnWidgetClass'
 
 provideComponent(ColumnContainerWidget, ({ widget }) => {
   const columns = widget.get('columns')
   if (!columns.length) return null
 
-  const content = columns.map((columnWidget, index) => {
-    const colSize = columnWidget.get('colSize') || 1
-    const className = widget.get('disableResponsiveAdaption')
-      ? `col-${colSize}`
-      : `col-md-${colSize}`
-    return (
-      <div key={index} className={className}>
-        <ContentTag
-          content={columnWidget}
-          attribute="content"
-          className="h-100"
-        />
-      </div>
-    )
-  })
-
-  const classNames = ['row']
-
-  if (widget.get('alignment')) {
-    classNames.push(`align-items-${widget.get('alignment')}`)
-  } else {
-    classNames.push('align-items-start')
+  const alignment = widget.get('alignment') || 'start'
+  const isResponsive = !widget.get('disableResponsiveAdaption')
+  const isFlex = widget.get('layoutMode') === 'flex'
+
+  const classNames = [`align-items-${alignment}`]
+  const flexClassName = isResponsive ? 'd-md-flex' : 'd-flex'
+  classNames.push(isFlex ? flexClassName : 'row')
+
+  return (
+    <div className={classNames.join(' ')}>
+      {columns.map((columnWidget: ColumnWidgetInstance) => {
+        const Column = isFlex ? FlexColumn : GridColumn
+        return (
+          <Column
+            key={columnWidget.id()}
+            columnWidget={columnWidget}
+            isResponsive={isResponsive}
+            isStretch={alignment === 'stretch'}
+          />
+        )
+      })}
+    </div>
+  )
+})
+
+type ColumnProps = {
+  columnWidget: ColumnWidgetInstance
+  isResponsive: boolean
+  isStretch: boolean
+}
+
+const FlexColumn = connect(function FlexColumn({
+  columnWidget,
+  isResponsive,
+  isStretch,
+}: ColumnProps) {
+  const classNames = isResponsive ? ['my-md-0', 'm-2'] : ['mx-2']
+  if (columnWidget.get('flexGrow')) classNames.push('flex-grow-1')
```

Comment:
I noticed, that one "grow" column of two "grow" columns can take more space, than 50%. E.g. if the right grow column contains an image:

<img width="1338" alt="Screenshot 2024-08-06 at 10 08 03" src="https://github.com/user-attachments/assets/7ae1bf49-9780-42b9-844d-cd78dab4e9e8">

Is this expected? By looking at the editor I would have expected to take "max 50%" of the available space. Is this a wrong expectation?

Reply:
> Is this a wrong expectation?
Yes, this is a consequence of how flex-grow works by default. And yes, the visualization can/will be misleading in these cases. Flex display is a tool for advanced editors.